pydifact/parser.py

```diff
@@ -193,7 +193,12 @@ def convert_tokens_to_segments(
 
             # here we can be sure that the token value is normal "content"
             # first backfill empty strings for skipped component data (:::)
-            for i in range(1, empty_component_counter):
+            for i in range(
+                1,
+                empty_component_counter
+                if data_element
+                else empty_component_counter + 1,
+            ):
                 data_element.append("")
```

Comment:
This was not my choice of formatting. I had it all on one line. Black broke it up onto multiple lines.

Reply:
Yes, I know - black sometimes does not do everything perfect. Ther's much discussion about that topic. But below the line, it's better to have one common style - and for that black is perfect. So I decided to stick with it.